src/Stack/Storage/User.hs

```diff
@@ -30,9 +24,11 @@ module Stack.Storage
     , saveCompilerPaths
     , upgradeChecksSince
     , logUpgradeCheck
+    -- * Shared utils
+    , updateList
```

Comment:
Perhaps move `updateList` and `updateSet` to another module, since there're not really "owned" by the user database code and could be used for any database?

Reply:
Addressed in 36f93646fe18008a23827288b8466878c41b5986

Reply:
Great, LGTM (well, once hlint warnings resolved :))